### PR TITLE
Added the Heatmap Graph Component

### PR DESCRIPTION
**Pull request by nxanil**
@ronakmshah Please check the Heatmap Graph component & let me know if you have any issues with it.

### CODE REVIEW

**Comment by nxanil** [comment]
@ronakmshah 

I have fixed the warnings.
Here is the screenshot of the graph at my end:
screenshot from 2017-04-07 12-02-07

To run it at my end, I had just run the simulator scripts and then changed the "SourceNSG" to "snsg:ovs-238" as per the values in my local elastic instance.

That's it.

**Comment by nxanil** [comment]
@curran @ronakmshah 

I have added the color coding for Heatmap.

Just want to be sure that colors are correct mapped:
1) InSla: Green
2) OutSla: Red
3) Unmonitored: Yellow

Thanks,
Anil

File: public/configurations/visualizations/aar-flow-sla-heatmap.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (curran) [inline public/configurations/visualizations/aar-flow-sla-heatmap.json]: Nice job with the configuration structure.

File: src/components/Graphs/AbstractGraph.js [hunk 2]
```diff
@@ -89,6 +89,32 @@ export default class AbstractGraph extends React.Component {
...
+
+        if (!colorColumn && !defaultColumn)
+            return;
+
+        let domainData = d3.map(data, (d) => d[colorColumn || defaultColumn]).keys().sort();
+
+        let propColors = [];
+        domainData.map((d) => {
+            if(mapColors[d]) {
+              propColors.push(mapColors[d]);
```
Comment (curran) [inline src/components/Graphs/AbstractGraph.js]: Please try to be consistent with the indentation spacing.

File: src/components/Graphs/LineGraph/index.js [hunk 3]
```diff
@@ -73,7 +73,7 @@ export default class LineGraph extends XYGraph {
 
         let xAxisHeight       = xLabel ? chartHeightToPixel : 0;
         let legendWidth       = legend.show && linesData.length > 1 ? this.longestLabelLength(data, legendFn) * chartWidthToPixel : 0;
-        let xLabelWidth       = this.longestLabelLength(data, xLabelFn) * chartWidthToPixel;
+
```
Comment (curran) [inline src/components/Graphs/LineGraph/index.js]: Was this change intentional? Looks like this line should stay.

Comment (nxanil) [inline src/components/Graphs/LineGraph/index.js]: Actually, It was not used as of now, so was giving warning on the console. But yes might be required while trimming the long labels.

Comment (curran) [inline src/components/Graphs/LineGraph/index.js]: Aha I see. In that case, thanks for removing it!

**Comment by curran** [comment]
@ronakmshah @nxanil From my side this is ready to merge.

**Comment by ronakmshah** [comment]
Merging this code. Congrats @nxanil for the first merge into this repo. Well done!
